build-common/src/org/jetbrains/kotlin/incremental/storage/BasicMap.kt

```diff
@@ -18,111 +18,127 @@ package org.jetbrains.kotlin.incremental.storage
 
 import com.intellij.util.io.DataExternalizer
 import com.intellij.util.io.EnumeratorStringDescriptor
+import com.intellij.util.io.IOUtil
 import com.intellij.util.io.KeyDescriptor
 import org.jetbrains.annotations.TestOnly
 import org.jetbrains.kotlin.incremental.IncrementalCompilationContext
 import org.jetbrains.kotlin.utils.Printer
 import java.io.File
 
-abstract class BasicMap<K : Comparable<K>, V, StorageType : LazyStorage<K, V>>(
-    internal val storageFile: File,
-    protected val storage: StorageType,
-    protected val icContext: IncrementalCompilationContext,
-) {
-    protected val pathConverter
-        get() = icContext.pathConverter
+/** [PersistentStorage] that provides a few extra utility methods. */
+interface BasicMap<KEY, VALUE> : PersistentStorage<KEY, VALUE> {
 
-    fun clean() {
-        storage.clean()
+    /** Removes all entries. */
+    fun clear() {
+        keys.forEach { remove(it) }
     }
 
+    /**
+     * DEPRECATED: This method should be removed because
+     *   - The `memoryCachesOnly` parameter is not used (it is always `false`).
+     *   - There is currently no good use case for flushing. In fact, the current implementation of this method does nothing.
+     */
     fun flush(memoryCachesOnly: Boolean) {
-        storage.flush(memoryCachesOnly)
+        check(!memoryCachesOnly) { "Expected memoryCachesOnly = false but it is `true`" }
     }
 
-    // avoid unsynchronized close
-    fun close() {
-        storage.close()
+    /**
+     * Deletes [storageFile] or a group of files associated with [storageFile] (e.g., an implementation of [PersistentStorage] may use a
+     * [com.intellij.util.io.PersistentHashMap], which creates files such as "storageFile.tab", "storageFile.tab.len", etc.).
+     *
+     * Make sure the storage has been closed first before calling this method.
```

Comment:
We could, but this is a default method in an interface (not an implementation class), so I imagine it would require more than a few lines of code to add that check.

If you think it’s nicer to have that check, please feel free to add it whenever you work with this code.